### ISSUE TITLE
Java backend replaces "-" with "_" in path

### ISSUE THREAD

**Issue by osavaryb** [dafny-lang/dafny#622]
When code is spilled by the java backend  (`-compileTarget:java -spillTargetCode:3`) with a path provided, DafnyDriver.cs l.593 replaces "-" by "_" resulting in the file being saved in the wrong folder.

**Comment by osavaryb** [comment]
One solution would be to replace 593 and 594 by
            ```
baseName = baseName.Replace('-', '_');
dafnyProgramName = Path.Combine(Path.GetDirectoryName(dafnyProgramName), baseName + Path.GetExtension(dafnyProgramName));
wr.WriteLine($"public class {baseName} {{");
```
so that '-' are replaced by '_' only in the class (and file) name rather than in the whole path

**Comment by davidcok** [comment]
Fixed in PR#640. The fix replaces any non-alphanumeric character (except _ and $) in a file name with '_' to form a class name. Note that this can cause name collisions if a user is prone to using punctuation or operators in file names.
